src/docfx/build/page/BookmarkValidator.cs

```diff
@@ -38,14 +38,14 @@ public List<(Error error, Document file)> Validate()
         {
             var result = new List<(Error error, Document file)>();
 
-            foreach (var (file, reference, bookmark) in _references)
+            foreach (var (file, reference, bookmark, range) in _references)
             {
                 if (_bookmarksByFile.TryGetValue(reference, out var bookmarks) && bookmarks.Contains(bookmark))
                 {
                     continue;
                 }
-                result.Add(file == reference ? (Errors.InternalBookmarkNotFound(file, bookmark, bookmarks), file) :
-                                               (Errors.ExternalBookmarkNotFound(file, reference, bookmark, bookmarks), file));
+                result.Add(file == reference ? (Errors.InternalBookmarkNotFound(file, bookmark, bookmarks).WithRange(range), file) :
```

Comment:
nit: if the InternalBookmarkNotFound method takes an `in Range` then you don't need `WithRange` to create a copy of `Error`